src/shared/components/case-viewer/case-review-specific-access-request/case-review-specific-access-request.component.ts

```diff
@@ -0,0 +1,121 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { CaseView, ErrorMessage, ReviewSpecificAccessRequest } from '../../../domain';
+import { AccessReason, ReviewSpecificAccessRequestErrors, ReviewSpecificAccessRequestPageText } from './models';
+import { AbstractAppConfig } from '../../../../app.config';
+
+@Component({
+  selector: 'ccd-case-review-specific-access-request',
+  templateUrl: './case-review-specific-access-request.component.html',
+  styleUrls: ['./case-review-specific-access-request.component.scss']
+})
+export class CaseReviewSpecificAccessRequestComponent
+  implements OnInit, OnDestroy {
+  public collapsed = false;
+  public title: string;
+  public hint: string;
+  public caseRefLabel: string;
+  public formGroup: FormGroup;
+  public submitted = false;
+  public errorMessage: ErrorMessage;
+  private readonly genericError = 'There is a problem';
+  private readonly radioSelectedControlName = 'radioSelected';
+  public readonly accessReasons: DisplayedAccessReason[];
+  public requestAccessDetails: RequestAccessDetails;
+  public caseSubscription: Subscription;
+  public userAccessType: string;
+  public caseDetails: CaseView;
+
+  constructor(
+    private readonly fb: FormBuilder,
+    private readonly route: ActivatedRoute,
+    private readonly appConfig: AbstractAppConfig,
+  ) {
+    this.accessReasons = [
+      { reason: AccessReason.APPROVE_REQUEST, checked: false },
+      { reason: AccessReason.REJECT_REQUEST, checked: false },
+      { reason: AccessReason.REQUEST_MORE_INFORMATION, checked: false },
+    ];
+  }
+
+  public ngOnInit(): void {
+    // TODO: this ticket is blocked so mocked with those data to go through, they will be removed and implimented with actual data
+    // when dependency resolved
+    this.setMockData();
+    this.title = ReviewSpecificAccessRequestPageText.TITLE;
+    this.hint = ReviewSpecificAccessRequestPageText.HINT;
+    this.caseRefLabel = ReviewSpecificAccessRequestPageText.CASE_REF;
+    this.formGroup = this.fb.group({
+      radioSelected: new FormControl(null, Validators.required),
+    });
+  }
+
+  public ngOnDestroy(): void {
+    if (this.caseSubscription) {
+      this.caseSubscription.unsubscribe();
+    }
+  }
+
+  public onChange(): void {
+    this.submitted = false;
+  }
+
+  public onSubmit(): void {
+    this.submitted = true;
+
+    if (this.formGroup.get(this.radioSelectedControlName).invalid) {
+      this.errorMessage = {
+        title: this.genericError,
+        description: ReviewSpecificAccessRequestErrors.NO_SELECTION,
+      };
+    }
+    // Initiate Review Access Request
+    if (this.formGroup.valid) {
```

Comment:
Just for info, not a lot going on here because we're still deciding on how to proceed with the Epic

Reply:
thanks for info